content/usgs-packages/sbtools_Discovery.Rmd

```diff
@@ -0,0 +1,224 @@
+
+## Discovering data via sbtools
+
+The ScienceBase search tools can be very powerful, but lack the ability to easily recreate the search. If you want to incorporate dataset queries into a reproducible workflow, you can script them using the `sbtools` query functions. The terminology differs from the web interface slightly. Below are functions available to query the catalog:
+
+1. `query_sb` (generic SB query)
+2. `query_sb_text` (matches title or description)
+3. `query_sb_doi` (use a DOI identifier)
+4. `query_sb_spatial` (data within or at a specific location)
+5. `query_sb_date` (items within time range)
+6. `query_sb_datatype` (type of data, not necessarily file type)
+
+These functions take a variety of inputs, and all return an R list of `sbitems` (a special `sbtools` class). All of these functions default to 20 returned search results, but you can change that by specifying the argument `limit`.  Before we practice using these functions, make sure you load the `sbtools` package in your current R session.
+
+```{r}
+library(sbtools)
+```
+
+### Using `query_sb`
+
+`query_sb` is the "catch-all" function for querying ScienceBase from R. It only takes one argument for specifying query parameters, `query_list`. This is an R list with specific query parameters as the list names and the user query string as the list values. See the `DESCRIPTION` section of the help file for all options (`?query_sb`). 
+
+```{r}
+
+##### THESE FIRST TWO SEARCHES STILL NEED WORK #####
+
+# search by keyword
+precip_query <- list(q = 'precipitation')
+precip_data <- query_sb(query_list = precip_query)
+length(precip_data) # 50 entries, so there is likely more than 50 results
+head(precip_data, 2)
+
+# search by keyword + category
+precip_maps_query <- list(q = 'precipitation', browseType = "Static Map Image", sort='title')
+precip_maps_data <- query_sb(query_list = precip_query)
+head(precip_maps_data, 2)
+
+##### -------------------------------------- #####
+
+# search by 2 keywords
+hazard2_query <- list(q = 'flood', q = 'earthquake')
+hazard2_data <- query_sb(query_list = hazard2_query)
+length(hazard2_data)
+head(hazard2_data, 2)
+
+# search by 3 keywords
+hazard3_query <- list(q = 'flood', q = 'earthquake', q='drought')
+hazard3_data <- query_sb(query_list = hazard3_query)
+length(hazard3_data)
+head(hazard3_data, 2)
+```
+
+### Using `query_sb_text`
+
+`query_sb_text` returns a list of `sbitems` that match the title or description fields. Use it to search authors, station names, rivers, states, etc.
+
+```{r}
+# search using a contributors name
+contrib_results <- query_sb_text("Robert Hirsch")
+head(contrib_results, 2)
+
+# search using place of interest
+park_results <- query_sb_text("Yellowstone")
+head(park_results, 2)
+
+# search using a site location
+loc_results <- query_sb_text("Embudo")
```

Comment:
Hmmm that's very true. All of these get handled in Science Base the same way I think, right? Since it's just searching text. Could do `river_results <- query_sb_text("Rio Grande")` to show another slightly different example.